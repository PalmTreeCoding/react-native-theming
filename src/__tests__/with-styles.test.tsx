import React from 'react'
import { View } from 'react-native'
import { render } from '@testing-library/react-native'

import { ThemeProvider, withStyles } from '../'

describe('withStyles', () => {

  const FakeComponent: React.FC = ({ styles }) => (
    <View style={styles.container} />
  )

  it('should add static styles to the component', () => {
    const styles = {
      container: {
        backgroundColor: 'pink'
      }
    }

    const Component = withStyles(styles)(FakeComponent)

    // @ts-ignore
    const { toJSON } = render(
      <ThemeProvider theme={{ customBackgroundColor: 'green' }}>
        <Component />
      </ThemeProvider>
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should add themed styles to the component', () => {
    const styles = (theme) => ({
      container: {
        backgroundColor: theme.customBackgroundColor
      }
    })

    const Component = withStyles(styles)(FakeComponent)

    // @ts-ignore
    const { toJSON } = render(
      <ThemeProvider theme={{ customBackgroundColor: 'green' }}>
        <Component />
      </ThemeProvider>
    )

    expect(toJSON()).toMatchSnapshot()
  })

})
