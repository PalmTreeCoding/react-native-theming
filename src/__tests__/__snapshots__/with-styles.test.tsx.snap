// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withStyles should add static styles to the component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "pink",
    }
  }
/>
`;

exports[`withStyles should add themed styles to the component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "green",
    }
  }
/>
`;
